### ISSUE TITLE
False positive ERROR [feature]: Not Set, when I use cargo test --features=[feature]

### ISSUE THREAD

**Issue by Razzwan** [rust-lang/rust#148142]
I am using module marked as feature, which I use only for tests

```rust
#[cfg(any(test, feature = "test_only"))]
pub mod test_only {
  // h
}
```
And then I run tests like this:
```bash
cargo test --features=test_only -- --test-threads=1
```

My integration tests (related on db) looks like this:
```rust
use api::test_only::*;
use std::any::Any;

#[tokio::test]
async fn test_1() -> Result<(), Box<dyn Any + Send + 'static>> {/* test 1 here */}

#[tokio::test]
async fn test_2() -> Result<(), Box<dyn Any + Send + 'static>> {/* test 2 here */}

#[tokio::test]
async fn test_3() -> Result<(), Box<dyn Any + Send + 'static>> {/* test 3 here */}
```

I expected to see all passed tests, like it was I previous rust versioin, but after update to rust 1.90.0

Instead, this happened: first test passed without any problems, but all the rest tests contain error:
```bash
test test_1 ... ok
test test_2 ... 2025-10-26T15:01:53.797196Z ERROR api::test_only: Not Set
ok
test it_game_partially_filled_removed_after_timeout ... 2025-10-26T15:01:54.985595Z ERROR game_api::test_only: Not Set
ok
test test_3 ... 2025-10-26T15:01:55.887607Z ERROR api::test_only: Not Set
ok

```

### Meta

`rustc --version --verbose`:
```
rustc 1.90.0 (1159e78c4 2025-09-14)
binary: rustc
commit-hash: 1159e78c4747b02ef996e55082b704c09b970588
commit-date: 2025-09-14
host: x86_64-unknown-linux-gnu
release: 1.90.0
LLVM version: 20.1.8
```

**Comment by fmease** [comment]
Could you please provide a complete reproducer? The snippets you've given don't provide enough context (at least to me personally) to figure out how to reproduce it or what is going on in general.

For example, you seem to be using some sort of logger / tracer judging by the lines like `2025-10-26T15:01:53.797196Z ERROR api::test_only: Not Set` which you didn't expect.

I tried the following which is obviously vastly insufficient: 

```toml
[package]
name = "hmm"
version = "0.1.0"
edition = "2024"

[features]
test_only = []

[dependencies]
tokio = { version = "1.48.0", features = ["rt", "macros", "tracing"] }
tokio-macros = "2.6.0"
```

```rs
#[cfg(any(test, feature = "test_only"))]
mod test_only {}

use test_only::*;

#[tokio::test]
async fn test_1() -> Result<(), Box<dyn std::any::Any + Send + 'static>> {
    Ok(())
}

#[tokio::test]
async fn test_2() -> Result<(), Box<dyn std::any::Any + Send + 'static>> {
    Ok(())
}

#[tokio::test]
async fn test_3() -> Result<(), Box<dyn std::any::Any + Send + 'static>> {
    Ok(())
}
```
